ThunderCloud/StormLink.swift

```diff
@@ -12,239 +12,266 @@ import Foundation
+                linkClass = .internal
+            } else if host == "native" {
+                linkClass = .native
+            }
+            break
+        case "mailto":
+            linkClass = .email
+            break
+        case "sms":
+            linkClass = .sms
+            break
+        case "tel":
+            linkClass = .call
+            break
+        default:
+            break
+        }
+    }
+    
+    /// Initialises a link to a storm page
+    ///
+    /// - Parameter pageId: The id of the storm page to link to
+    public init?(pageId: String) {
+        
+        let pageURL: URL?
+        
+        if let metadata = ContentController.shared.metadataForPage(withId: pageId), let src = metadata["src"] as? String {
+            
+            pageURL = URL(string: src) ?? URL(string: "cache://pages/\(pageId).json")
+            
+        } else {
+            
+            pageURL = URL(string: "cache://pages/\(pageId).json")
+        }
+        
+        guard let _pageURL = pageURL else {
+            return nil
+        }
+        
+        title = "Link"
+        url = _pageURL
+        linkClass = .internal
+        body = nil
+        recipients = nil
+        appIdentityIdentifier = nil
+        destination = nil
+        duration = nil
+    }
+    
+    /// Initialises a link to a storm page
+    ///
+    /// - Parameter pageName: The internal name of the storm page to link to
+    public init?(pageName: String) {
+        
+        guard let metadata = ContentController.shared.metadataForPage(withName: pageName), let src = metadata["src"] as? String else {
+            return nil
+        }
+        
+        guard let srcURL = URL(string: src) else { return nil }
+        
+        title = "Link"
+        url = srcURL
+        linkClass = .internal
+        body = nil
+        recipients = nil
+        appIdentityIdentifier = nil
+        destination = nil
+        duration = nil
+    }
+    
+    public override init() {
+        
+        title = "Link"
+        url = nil
+        linkClass = .internal
+        body = nil
+        recipients = nil
+        appIdentityIdentifier = nil
+        destination = nil
+        duration = nil
+    }
+    
+    //MARK: - Standard link properties
+    
+    /// The title to describe the link
+    public var title: String?
+    
+    /// The URL of the link
+    public var url: URL?
+    
+    /// The type of link
+    ///
+    /// Storm has various link types for different link behaviours. They are represented as different objects in the CMS, but ultimately are represented by the same model natively
+    public var linkClass: LinkClass
+    
+    //MARK: - SMS and Email -
+    
+    /// The body of the text to be shared
+    ///
+    /// Only valid for Email/SMS links
+    public var body: String?
+    
+    /// An array of recipients that the body should be shared to
+    ///
+    /// Only valid for SMS links
+    public var recipients: [String]?
```

Comment:
@simonmitchell many previous `let`s here have since been changed to use `var`. This hasn't caused any of the existing tests to fail, yet may be something to be aware of!